File: tests/fixtures/sample.ts
// tests/fixtures/sample.ts

/** Greets a user by name */
export function greet(name: string): string {
  return `Hello, ${name}!`;
}

export class UserService {
  /** Creates a new user */
  createUser(name: string, email: string): User {
    const validated = validateEmail(email);
    return { name, email: validated };
  }

  getUser(id: number): User | null {
    return null;
  }
}

function validateEmail(email: string): string {
  if (!email.includes("@")) {
    throw new Error("Invalid email");
  }
  return email.toLowerCase();
}

interface User {
  name: string;
  email: string;
}

File: tests/fixtures/sample.py
# tests/fixtures/sample.py

"""Sample Python module for testing."""

import os
from pathlib import Path


def greet(name: str) -> str:
    """Greets a user by name."""
    return f"Hello, {name}!"


class UserService:
    """Handles user operations."""

    def create_user(self, name: str, email: str) -> dict:
        """Creates a new user."""
        validated = validate_email(email)
        return {"name": name, "email": validated}

    def get_user(self, user_id: int) -> dict | None:
        return None


def validate_email(email: str) -> str:
    if "@" not in email:
        raise ValueError("Invalid email")
    return email.lower()
